include/sparrow/arrow_interface/arrow_array_schema_utils.hpp

```diff
@@ -0,0 +1,264 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <optional>
+#include <ranges>
+#include <tuple>
+#include <type_traits>
+#include <vector>
+
+#include "sparrow/memory.hpp"
+#include "sparrow/mp_utils.hpp"
+
+namespace sparrow
+{
+    /**
+     * Get the size of a range, a tuple or an optional.
+     * If the range is a sized range, the size is obtained by calling size().
```

Comment:
```suggestion
     * If the range is a sized range, the size is obtained by calling `std::ranges::size()`.
```

That function doesnt necessarilly call `.size()`, it depends on the kind of range, so we better be specific. For example if T is a normal array, there is no `size` member, but it will still work.

Reply:
Also note that you can use anti-quotes around name which are code, that will help with the generated documentation.

Reply:
Ok done